### PR TITLE
#1741 [Create application] Unregistered user wants to create an application to workshop

### PR DESCRIPTION
**Pull request by inga-lekhman**

### CODE REVIEW

File: src/app/shared/components/workshop-modal/workshop-modal.component.html [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      class="workshop-modal__action-button white"
+      mat-raised-button
+      mat-dialog-close
+    >
+      Повернутись
+    </button>
+    <button
+      *ngIf="data.buttonLabel"
+      class="workshop-modal__action-button"
+      mat-raised-button
```
Comment (litvinets) [inline src/app/shared/components/workshop-modal/workshop-modal.component.html:24]: fix styles please, I belive you should add shared button styles to the scss file
image
besides color, both button should have teh same width

File: src/app/shell/details/side-menu/actions/actions.component.html [hunk 2]
```diff
@@ -2,9 +2,9 @@
   <h4 class="card-title" *ngIf="workshop.price" fxFlexAlign="center"> {{ workshop.price }} грн/{{ PayRateTypeUkr[workshop.payRate] }} </h4>
   <h4 class="card-title" *ngIf="!workshop.price"> Безкоштовний гурток </h4>
     <button mat-raised-button class="button button-application" fxLayout="row" fxLayoutAlign="center center" [disabled]="hideApplicationSubmission">
-      <a 
+      <a
```
Comment (litvinets) [inline src/app/shell/details/side-menu/actions/actions.component.html:5]: please, wrap teh button into this link attribute 
<a> -> <button>

File: src/app/shared/components/workshop-modal/workshop-modal.component.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  buttonLabel?: string;
+}
+
+@Component({
+  selector: 'app-workshop-modal',
+  templateUrl: './workshop-modal.component.html',
+  styleUrls: ['./workshop-modal.component.scss']
+})
+
+export class WorkshopModalComponent {
```
Comment (litvinets) [inline src/app/shared/components/workshop-modal/workshop-modal.component.ts:18]: I believe it is not teh best naming. Obviously, this functionality is not related to the workshop at all, it can be applied everywhere for unregistered user. So, consider better naming

File: src/app/shared/components/workshop-modal/workshop-modal.component.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import {Component, Inject} from '@angular/core';
+import {MAT_DIALOG_DATA} from '@angular/material/dialog';
+import {Login} from '../../store/registration.actions';
+import {Store} from '@ngxs/store';
+
+interface DialogData {
```
Comment (litvinets) [inline src/app/shared/components/workshop-modal/workshop-modal.component.ts:6]: move it to modals folder

**Review by litvinets — CHANGES_REQUESTED** [review]

File: src/app/shared/components/workshop-modal/workshop-modal.component.html (hunk 1 above)
Comment (inga-lekhman) [reply src/app/shared/components/workshop-modal/workshop-modal.component.html:24]: it's not a styling issue, it's how autofocus works for modal, I'll add a property to prevent the buttons from being highlighted

File: src/app/shared/components/workshop-modal/workshop-modal.component.ts (hunk 3 above)
Comment (inga-lekhman) [reply src/app/shared/components/workshop-modal/workshop-modal.component.ts:18]: Sure, will change the naming. Thank you

**Review by litvinets — CHANGES_REQUESTED** [review]
There is a button "like workshop", it should open the modal with the text 'Щоб додати гурток в улюблені, зареєструйтеся на порталі. Дякуємо

'
image

**Review by litvinets — CHANGES_REQUESTED** [review]

**Comment by inga-lekhman** [comment]
@litvinets thank you for the catch. I missed that part with adding to favorite

**Review by litvinets — APPROVED** [review]
well done!
